apps/prairielearn/assets/scripts/instructorFileEditorClient.ts

```diff
@@ -98,6 +111,92 @@ class InstructorFileEditor {
         .querySelector<HTMLButtonElement>('.js-reformat-file')
         ?.addEventListener('click', () => this.reformatJSONFile());
     }
+
+    // Override the save button click to show confirmation modal if needed
+    this.saveElement!.addEventListener('click', (e) => this.handleSaveClick(e));
+  }
+
+  /**
+   * Handles the save button click, showing a confirmation modal if there are issues,
+   * or proceeding with the save if the user has already confirmed.
+   */
+  handleSaveClick(event: MouseEvent) {
+    if (this.confirmedSave) {
+      this.confirmedSave = false;
+      return;
+    }
+
+    const issues = this.checkForSaveIssues();
+
+    if (issues.length > 0) {
+      event.preventDefault();
+      this.showConfirmationModal(issues);
+    }
+
+    // Otherwise, continue with the save
+  }
+
+  /**
+   * Checks for issues that would require confirmation before saving.
+   *
+   * @returns An array of issues that would require confirmation before saving.
+   */
+  checkForSaveIssues(): string[] {
```

Comment:
The issues contained here seem to be mutually exclusive, it seems to me like we could simplify things (and make the UI a bit nicer) if we instead always return exactly one error message:

<img width="825" height="325" alt="Image" src="https://github.com/user-attachments/assets/dfbf173f-a06b-4591-a486-b8bb4c35ae2d" />

Reply:
Fixed.